File: app/src/main/java/java_version/scanner/QrScannerActivity.java
package java_version.scanner;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.widget.Toast;

import com.budiyev.android.codescanner.AutoFocusMode;
import com.budiyev.android.codescanner.CodeScanner;
import com.budiyev.android.codescanner.CodeScannerView;
import com.budiyev.android.codescanner.DecodeCallback;
import com.budiyev.android.codescanner.ScanMode;
import com.google.zxing.Result;
import com.spacepal.internal.app.BaseActivity;
import com.spacepal.internal.app.R;

import org.jetbrains.annotations.Nullable;

/**
 * Created by sidhu on 6/4/2018.
 */

public class QrScannerActivity extends BaseActivity implements DecodeCallback {
    private CodeScanner mCodeScanner;
    @Override
    public int getId() {
        return R.layout.activity_scanner;
    }

    @Override
    public void created(@Nullable Bundle savedInstanceState) {
        CodeScannerView scannerView = findViewById(R.id.scanner_view);
        mCodeScanner = new CodeScanner(this, scannerView);
        mCodeScanner.setAutoFocusEnabled(true);
        mCodeScanner.setAutoFocusMode(AutoFocusMode.SAFE);
        mCodeScanner.setScanMode(ScanMode.SINGLE);
        mCodeScanner.setDecodeCallback(this);
        scannerView.setOnClickListener(view -> mCodeScanner.startPreview());
    }

    @Override
    public void onDecoded(@NonNull Result result) {
        runOnUiThread(() -> {
            Toast.makeText(this, result.getText(), Toast.LENGTH_SHORT).show();
        });
    }

    @Override
    protected void onResume() {
        super.onResume();
        mCodeScanner.startPreview();
    }

    @Override
    protected void onPause() {
        mCodeScanner.releaseResources();
        super.onPause();
    }
}

File: app/src/main/java/com/spacepal/internal/app/ui/scanner/QrScannerActivity.kt
package com.spacepal.internal.app.ui.scanner

import android.os.Bundle
import android.widget.Toast
import com.budiyev.android.codescanner.*
import com.spacepal.internal.app.BaseActivity
import com.spacepal.internal.app.R

/**
 * Created by sidhu on 6/4/2018.
 */
class QrScannerActivity : BaseActivity() {

    override val id: Int
        get() = R.layout.activity_scanner

    override fun created(savedInstanceState: Bundle?) {
        val scannerView = findViewById<CodeScannerView>(R.id.scanner_view)

        codeScanner = CodeScanner(this, scannerView)

        // Parameters (default values)
        codeScanner.camera = CodeScanner.CAMERA_BACK // or CAMERA_FRONT or specific camera id
        codeScanner.formats = CodeScanner.ALL_FORMATS // list of type BarcodeFormat,
        // ex. listOf(BarcodeFormat.QR_CODE)
        codeScanner.autoFocusMode = AutoFocusMode.SAFE // or CONTINUOUS
        codeScanner.scanMode = ScanMode.SINGLE // or CONTINUOUS or PREVIEW
        codeScanner.isAutoFocusEnabled = true // Whether to enable auto focus or not
        codeScanner.isFlashEnabled = true // Whether to enable flash or not

        // Callbacks
        codeScanner.decodeCallback = DecodeCallback {
            runOnUiThread {
                Toast.makeText(this, "Scan result: " + it.text, Toast.LENGTH_LONG).show()
            }
        }
        codeScanner.errorCallback = ErrorCallback { // or ErrorCallback.SUPPRESS
            runOnUiThread {
                Toast.makeText(this, "Camera initialization error: " + it.message,
                        Toast.LENGTH_LONG).show()
            }
        }

        scannerView.setOnClickListener {
            codeScanner.startPreview()
        }    }

    private lateinit var codeScanner: CodeScanner




    override fun onResume() {
        super.onResume()
        codeScanner.startPreview()
    }

    override fun onPause() {
        codeScanner.releaseResources()
        super.onPause()
    }
}